api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java

```diff
@@ -204,27 +205,123 @@ public CohortDefinition getPatientsBasedOnPatientStatesBeforeDate(int program, i
    * @return CohortDefinition
    */
   public CohortDefinition getDeceasedPatients() {
-    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    SqlCohortDefinition cd = new SqlCohortDefinition();
     cd.setName("Get deceased patients based on patient states and person object");
-    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
-    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("onOrBefore", "End Date", Date.class));
     cd.addParameter(new Parameter("location", "Location", Location.class));
+    String sql =
+        "SELECT patient_id "
+            + "FROM   (SELECT patient_id, "
+            + "               Max(death_date) death_date "
+            + "        FROM   (SELECT p.patient_id, "
```

Comment:
There doesn't seem to be a closing bracket for this one...

Reply:
ah I got confused here, actually I see it on 278. Sorry got confused there :man_facepalming:

Reply:
Sorry I got a bit confused here, pls ignore :man_facepalming: I see the matching closing bracket on line 278